packetery/public/admin-country-carrier.js

```diff
@@ -0,0 +1,121 @@
+                    })
+                    .on('click', '.js-delete', function () {
+                        multiplier.deleteOption(this);
+                    })
+                    .each(function () {
+                        var $wrapper = $(this);
+                        var $inputs = $wrapper.find('input[type="number"]');
+                        // add only if there is something in last row
+                        if ($inputs.eq(-2).val() || $inputs.eq(-1).val()) {
+                            multiplier.addOption($wrapper.find('.js-add'), $wrapper, false);
+                        }
+                        multiplier.toggleDeleteButton($wrapper);
+                    });
+            };
+
+            this.addOption = function (button, $wrappers, setFocus) {
+                var wrapperClassName = $wrappers.first().attr('class'),
+                    $wrapper = $(button).closest('.' + wrapperClassName),
+                    $template = getTemplate($wrapper);
+
+                updateIds($template, newId++);
+                $wrapper.find('table').append($template);
+                if (typeof setFocus === 'undefined') {
+                    setFocus = true;
+                }
+                if (setFocus) {
+                    $('input', $template).eq(0).focus();
+                }
+                this.toggleDeleteButton($wrapper);
+            };
+
+            this.deleteOption = function (button) {
+                var $row = $(button).closest('tr'),
+                    $table = $row.closest('table');
+
+                $row.remove();
+                this.toggleDeleteButton($table);
+            };
+
+            this.toggleDeleteButton = function ($wrapper) {
+                var optionsCount = $wrapper.find('tr').length,
+                    $buttons = $wrapper.find('button.js-delete');
+
+                (optionsCount > 1) ? $buttons.show() : $buttons.hide();
+            };
+
+            /**
+             * Find the highest counter in the rendered form (invalid form gets re-rendered with its submitted new_* form items)
+             */
+            function findMaxNewId() {
+                var $newInputs = $('[name*=' + prefix + ']'),
+                    maxNewId = 1;
+
+                $newInputs.each(function () {
+                    var newIdMatch = $(this).attr('name').match('\\[' + prefix + '(\\d+)\\]');
+                    var counter = parseInt(newIdMatch[1]);
+                    maxNewId = Math.max(maxNewId, counter + 1);
+                });
+
+                return maxNewId;
+            }
+
+            var prefix = 'new_',
+                newId = findMaxNewId();
+
+            function getTemplate($wrapper) {
+                var $template = $wrapper.find('.js-template').clone().removeClass('js-template');
+                $template.find('input').val('').removeAttr('required').removeAttr('data-nette-rules');
```

Comment:
Todle je divné. Mělo by se to vymyslet jinak. v data-nette-rules jsou validační pravida. To jestli se políčko má validovat by měl určovat pouze backend (nette form).

Reply:
Podívám se na to, až probereme chování inputů.

Reply:
Upraveno.